thorlcr/master/thgraphmanager.cpp

```diff
@@ -1109,31 +1111,89 @@ void thorMain(ILogMsgHandler *logHandler, const char *wuid, const char *graphNam
         Owned<CJobManager> jobManager = new CJobManager(logHandler);
         try
         {
-            if (wuid) // one-shot, quits after running
+#ifndef _CONTAINERIZED
+            jobManager->run();
+#else
+            unsigned lingerPeriod = globals->getPropInt("@lingerPeriod", DEFAULT_LINGER_SECS*1000);
```

Comment:
That is true, but and end of job it does signal to the Thor(s) and terminates them.
So they (or 1 if not parallel workflow) linger whilst job is running, reuse the 1 if lingering, and shut it down when job is finished (when there are no more graphs coming).

if you knew it really was last graph (no conditional possible graphs to come), then yes a 'thisislast' hint would be useful.

Reply:
Agreed.  The problem isn't if the job terminates, it is if it is still running.  It will block this thor for the linger period.    I suspect a much smaller linger period might be a better default.

Reply:
Reducing to 10s